django/applications/catmaid/static/js/widgets/tagtable.js

```diff
@@ -0,0 +1,365 @@
+                checked: false
+              };
+            });
+
+            var table = $(tableSelector).DataTable();
+
+            table.rows.add(rowObjs);
+            table.draw();
+
+            $(tableSelector + '_processing').hide();
+          }
+        );
+      },
+      init: function() {
+        this.init(project.getId());
+      }
+    };
+  };
+
+  /**
+   * Update the skeleton source the widget uses to export skeletons
+   *
+   * @param {Object} obj - skeleton IDs to add or subtract from skeleton source
+   * @param {number[]} obj.add - array of skeleton IDs to add to skeleton source
+   * @param {number[]} obj.subtract - array of skeleton IDs to subtract from skeleton source
+   */
+  TagTable.prototype.addAndSubtractFromSkeletonSource = function(obj) {
+    // Update the skeleton source run by the widget
+    this.selectedSkeletons.append(
+      obj.add.reduce(function (mappingObj, currentID) {
+        mappingObj[currentID] = new CATMAID.SkeletonModel(currentID);
+        return mappingObj;
+      }, {})
+    );
+    this.selectedSkeletons.removeSkeletons(obj.subtract);
+  };
+
+  Set.prototype.difference = function(otherSet) {  // separate function?
+    var difference = new Set(this);
+    otherSet.forEach(function(elem) {
+      difference.delete(elem);
+    });
+    return difference;
+  };
+
+  Set.prototype.addAll = function(array) {  // separate function?
+    for (var i = 0; i < array.length; i++) {
+      this.add(array[i]);
+    }
+    return this;
+  };
+
+  var createCheckbox = function(checked, row) {
+    var checkbox = $('<input />', {
+        type:'checkbox',
+        class:'skelSelector',
+        id: 'skelSelector' + row.id,
+        value: row.id,
+        checked: checked
+      });
+
+    return checkbox.prop('outerHTML');
+    };
+
+  TagTable.prototype.init = function() {
+    var widgetID = this.widgetID;
+    var tableSelector = "#tagtable" + widgetID;
+
+    this.oTable = $(tableSelector).dataTable({  // use ajax data source directly?
+      // http://www.datatables.net/usage/options
+      "bDestroy": true,
+      "sDom": '<"H"lrp>t<"F"ip>',
+      "bServerSide": false,
+      "paging": true,
+      "bLengthChange": true,
+      "bAutoWidth": false,
+      "iDisplayLength": 30,
+      "aLengthMenu": [CATMAID.pageLengthOptions, CATMAID.pageLengthLabels],
+      "bJQueryUI": true,
+      "processing": true,
+      "deferRender": true,
+      "columns": [
+        {
+          "data": 'id',
+          "orderable": true,
+          "searchable": true,
+          "className": "center"
+        },
+        {
+          "data": 'tag',
+          "orderable": true,
+          "searchable": true,
+          "className": "center"
+        },
+        {
+          "data": 'skeletons',
+          "orderable": true,
+          "className": "center"
+        },
+        {
+          "data": 'checked',
+          "render": function(data, type, row) { return createCheckbox(data, row); },
```

Comment:
You might be able to improve sorting performance if you only create a checkbox if `type === 'display'` (see https://datatables.net/reference/option/columns.render). And for other types it would probably fine to just `return data` or maybe  `data ? 1 : 0` since the documentation says it should be numeric or a string for sorting.